src/mahoji/lib/abstracted_commands/nightmareZoneCommand.ts

```diff
@@ -0,0 +1,479 @@
+	const quantity = Math.floor(maxTripLength / timePerMonster);
+	const duration = quantity * timePerMonster;
+	// Consume GP (and prayer potion if experience setup)
+	const dreamCost = new Bank().add(
+		'Coins',
+		Math.floor(((QP >= MAX_QP ? 16_000 : 26_000) * duration) / ((strategy === 'points' ? 30 : 60) * Time.Minute))
+	);
+	if (strategy === 'experience') {
+		dreamCost.add('Prayer potion(4)', Math.floor(duration / (Time.Minute * 5)));
+	}
+	const totalCost = new Bank(dreamCost).clone();
+
+	if (!user.owns(totalCost)) {
+		return `You don't have the required gold coins for this trip. You need ${totalCost}.`;
+	}
+
+	await user.removeItemsFromBank(totalCost);
+	updateBankSetting('nmz_cost', totalCost);
+	await trackLoot({
+		id: 'nmz',
+		type: 'Minigame',
+		totalCost,
+		changeType: 'cost',
+		users: [
+			{
+				id: user.id,
+				cost: totalCost
+			}
+		]
+	});
+
+	await addSubTaskToActivityTask<NightmareZoneActivityTaskOptions>({
+		quantity,
+		userID: user.id,
+		duration,
+		type: 'NightmareZone',
+		channelID: channelID.toString(),
+		minigameID: 'nmz',
+		strategy
+	});
+
+	return `${
+		user.minionName
+	} is now killing ${quantity}x monsters in the Nightmare Zone! It will take ${formatDuration(
+		duration
+	)} to finish. **Boosts:** ${boosts.join(', ')}\nYour minion used up ${totalCost}`;
+}
+
+export async function nightmareZoneShopCommand(
+	interaction: ChatInputCommandInteraction,
+	user: MUser,
+	item: string | undefined,
+	quantity = 1
+) {
+	const currentUserPoints = user.user.nmz_points;
+	if (!item) {
+		return `You currently have ${currentUserPoints.toLocaleString()} Nightmare Zone points.`;
+	}
+
+	if (user.user.minion_ironman) {
+		return `${user.usernameOrMention} is an ironman, so they can't buy anything from this shop!`;
+	}
+
+	const shopItem = nightmareZoneBuyables.find(
+		i => stringMatches(item, i.name) || i.aliases.some(alias => stringMatches(alias, item))
+	);
+	if (!shopItem) {
+		return `This is not a valid item to buy. These are the items that can be bought using Nightmare Zone point: ${nightmareZoneBuyables
+			.map(v => v.name)
+			.join(', ')}`;
+	}
+
+	const cost = quantity * shopItem.cost;
+	if (cost > currentUserPoints) {
+		return `You don't have enough Nightmare Zone points to buy ${quantity.toLocaleString()}x ${shopItem.name} (${
+			shopItem.cost
+		} Nightmare Zone points each).\nYou have ${currentUserPoints} Nightmare Zone points.\n${
+			currentUserPoints < shopItem.cost
+				? "You don't have enough Nightmare Zone points for any of this item."
+				: `You only have enough for ${Math.floor(currentUserPoints / shopItem.cost).toLocaleString()}`
+		}`;
+	}
+
+	await handleMahojiConfirmation(
+		interaction,
+		`Are you sure you want to spend **${cost.toLocaleString()}** Nightmare Zone points to buy **${quantity.toLocaleString()}x ${
+			shopItem.name
+		}**?`
+	);
+
+	await transactItems({
+		userID: user.id,
+		collectionLog: true,
+		itemsToAdd: new Bank(shopItem.output).multiply(quantity)
```

Comment:
do the way we normally do:
`const loot = new Bank()`
`Are you sure you want to spend ${x} to buy ${loot}?`